### PR TITLE
DOCSP-44608: merge v2 into master

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-php-library/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-44608>
Staging - 

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x]  Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -74,16 +74,43 @@ Detailed installation instructions may be found in the
...
 
-For more information on release changes, see the release notes and associated 
-JIRA tickets for each release on `GitHub <https://github.com/mongodb/mongo-php-library/releases>`__.
+For more information on release changes, see the release notes and
+associated JIRA tickets for each release on `GitHub
+<https://github.com/mongodb/mongo-php-library/releases>`__.
+
+.. _php-v2-breaking-changes:
+
+Version 2.0 Breaking Changes
```
Comment (jmikola) [inline source/upgrade.txt:92]: I find it strange the `whats-new.txt` links to this page as a source of more information, but it actually contains less content. Is there any reason to have separate pages?

File: source/reference/method/MongoDBDatabase-listCollectionNames.txt [hunk 2]
```diff
@@ -74,8 +74,8 @@ Parameters
 Return Values
 -------------
 
-An :php:`Iterator <class.iterator.php>`, which provides the name of each
-collection in the database.
+An :php:`Iterator <manual/en/class.iterator.php>`, which provides the
+name of each collection in the database.
```
Comment (jmikola) [inline source/reference/method/MongoDBDatabase-listCollectionNames.txt:78]: I like this wording more than "contains" or "stored in". I expect this was just written at a different time than the other pages, which is why they're inconsistent.

Saying that an "iterator provides something" is a concise way to convey the technical aspect of it yielding results when traversed (in simpler terms).

File: source/reference/method/MongoDBCollection-findOne.txt [hunk 3]
```diff
@@ -27,154 +25,7 @@ Definition
...
-     * - sort
-       - array|object
-       - The sort specification for the ordering of the results.
-
-     * - typeMap
-       - array
-       - .. include:: /includes/extracts/collection-option-typeMap.rst
-
-         This will be used for the returned result document.
+.. include:: /includes/reference/find-one-parameters.rst
```
Comment (jmikola) [inline source/reference/method/MongoDBCollection-findOne.txt:28]: As with the `find` include, please confirm that this include exactly matches the content from the corresponding methods (e.g. no `versionadded` inconsistencies).

File: source/reference/class/MongoDBModelCollectionInfo.txt [hunk 4]
```diff
@@ -8,8 +8,8 @@ Definition
 .. phpclass:: MongoDB\Model\CollectionInfo
 
    This class models information about a collection. Instances of this class are
-   returned by traversing a :phpclass:`MongoDB\Model\CollectionInfoIterator`,
-   which is returned by :phpmethod:`MongoDB\Database::listCollections()`.
+   stored in an :php:`Iterator <manual/en/class.iterator.php>`
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/reference/class/MongoDBModelCollectionInfo.txt:11]: I can preserve the original language

File: source/reference/method/MongoDBCollection-dropSearchIndex.txt [hunk 5]
```diff
@@ -4,8 +4,6 @@ MongoDB\\Collection::dropSearchIndex()
 
 .. versionadded:: 1.17
 
-.. default-domain:: mongodb
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/reference/method/MongoDBCollection-dropSearchIndex.txt:7]: I've only removed this from the pages that I've touched in this PR to avoid cluttering the "Files Changed"

File: source/reference/method/MongoDBCollection-find.txt [hunk 6]
```diff
@@ -27,203 +25,7 @@ Definition
...
-       - Prevents the cursor from returning a document more than once because of
-         an intervening write operation.
-
-         .. deprecated:: 1.4
-         .. versionadded:: 1.2
-
-     * - typeMap
-       - array
-       - .. include:: /includes/extracts/collection-option-typeMap.rst
+.. include:: /includes/reference/find-parameters.rst
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/reference/method/MongoDBCollection-find.txt:28]: I can double check

[the remark this answers is not in the document]
Comment (rustagir) [reply source/reference/method/MongoDBCollection-find.txt:28]: Ah, they have differences in the extracts that they use, which use different language for bucket vs. collection. If this seems like an important difference to preserve, I will revert from using the includes. I'll wait on your reply to make changes

File: source/whats-new.txt [hunk 7]
```diff
@@ -28,6 +30,84 @@ following versions of the {+php-library+}:
...
+
+The {+library-short+} v2.0 release includes the following features,
+improvements, and fixes:
+
+- Removes deprecated fields in GridFS types.
+
+  -  The library does not calculate the ``md5`` field when a file is
+     uploaded to GridFS. If your application requires a file digest, you
+     must implement this process outside GridFS and store the values in
+     metadata.
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/whats-new.txt:64]: Sure, created https://jira.mongodb.org/browse/DOCSP-44698

File: source/whats-new.txt [hunk 8]
```diff
@@ -28,6 +30,84 @@ following versions of the {+php-library+}:
...
+       ``dropIndexes()``, ``dropSearchIndex()``, ``rename()``
+     - ``MongoDB\Database``: ``createCollection()``, ``drop()``,
+       ``dropCollection()``, ``renameCollection()``
+   
+   To learn more about breaking changes in this release, see the
+   :ref:`Version 2.0 Breaking Changes <php-v2-breaking-changes>` section
+   in the Upgrade Library Versions guide.
+
+The {+library-short+} v2.0 release includes the following features,
+improvements, and fixes:
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/whats-new.txt:57]: This is just a generic intro sentence, i'll revise to be more consistent

File: source/upgrade.txt (hunk 1 above)
Comment (rustagir) [reply source/upgrade.txt:92]: We only use the upgrade guide to address **breaking changes**, whereas the whats new guide captures all changes. the upgrade guids can help users learn more about the specific steps they must take to address the breaking change instead of just describing the change

File: source/reference/method/MongoDBCollection-findOne.txt (hunk 3 above)
Comment (rustagir) [reply source/reference/method/MongoDBCollection-findOne.txt:28]: same issue as above with the extracts

File: source/reference/method/MongoDBCollection-dropSearchIndex.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmikola) [reply source/reference/method/MongoDBCollection-dropSearchIndex.txt:7]: If it's not actually related to this PR would it make sense to do another that removes it everywhere in one shot?

File: source/reference/method/MongoDBCollection-find.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (jmikola) [reply source/reference/method/MongoDBCollection-find.txt:28]: That might be what I was thinking of. I think it makes sense to revert and keep what he had in place before (with the necessary changes for 2.0).

File: source/whats-new.txt [hunk 9]
```diff
@@ -65,7 +65,7 @@ improvements, and fixes:
 
   - The fields ``contentType`` and ``aliases`` are no longer stored in
     the ``files`` GridFS collection. If your application requires this
-    information, you must store it in metadata. To learn mor about
+    information, you must store it in metadata. To learn More about
```
Comment (jmikola) [inline source/whats-new.txt:68]:
```suggestion
    information, you must store it in metadata. To learn more about
```

**Review by jmikola — APPROVED** [review]
LGTM with capitalization fix.

File: source/reference/method/MongoDBCollection-dropSearchIndex.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (rustagir) [reply source/reference/method/MongoDBCollection-dropSearchIndex.txt:7]: I can do that after merging this, sure!

File: source/whats-new.txt (hunk 9 above)
Comment (rustagir) [reply source/whats-new.txt:68]: ack! ty
